src/projections/airocean.cpp

```diff
@@ -0,0 +1,345 @@
+
+
+// By default the resulting orientation of the projection is vertical
+// the following transforms are used to alter the projection data
+// so that the resulting orientation is horizontal instead
+constexpr double orient_horizontal_trans[4][4] = {{0.0, -1.0, 0.0, 36843762.068421006}, {1.0, 0.0, 0.0, 0.0}, {0.0, 0.0, 1.0, 0.0}, {0.0, 0.0, 0.0, 1.0}};
```

Comment:
Isn't the value 36843762.068421006 specific to the use of the GRS80 ellipsoid ?
And I suspect that P->left and P->right shouldn't be overridden, and P->a not used directly, to let the generic code in src/fwd.cpp and src/inv.cpp do the scaling from the unit ellipsoid to the target one.

Reply:
@plouvart gentle ping w.r.t my above question

Reply:
Hello @rouault 
Sorry for the lack of response over the last few week's. Lots of personal and work related obligations.
I will have time for a closer look at the issue this week end.

Reply:
I remember that I got the inspiration for the P->right, P->left and P->a lines from the [s2 projection file](https://github.com/plouvart/PROJ/blob/18f9d9eafa6693a32ae0391501b174b79e70345b/src/projections/s2.cpp#L429)
I didn't know that src/fwd and src/inv does that for us automatically. I can remove those lines.

About the 36843762.068421006 value, it is related to the final scale of the airocean space whose dimensions stays constant whatever ellipsoid model we use (since it all gets _normalized_ to the unit sphere in the end anyway).
Unless I'm mistaken?

Reply:
> About the 36843762.068421006 value, it is related to the final scale of the airocean space whose dimensions stays constant whatever ellipsoid model we use (since it all gets _normalized_ to the unit sphere in the end anyway).

sounds surprising to me that the extent of the projection is constant and not dependent on the ellipsoid semi-major axis, but I don't know anything about that projection. I guess a could way to check that would be to add round-trip tests with the unit ellipsoid

Reply:
> I guess a could way to check that would be to add round-trip tests with the unit ellipsoid
That's a good idea. I'll try that

Reply:
Round trip tests work when the ellipsoid/no ellipsoid is the same for fwd and inv successively.
Obviously choosing a different ellipsoid model for the fwd and inv model result in a difference.

The dimensions of the Airocean space are determined by the length of the edges of the triangle faces of the icosahedron, which depends on the arbitrary choice of the radius of the sphere it is circumscribed within.
The choice was made to have that scaling factor equals the arithmetic mean radius of the earth, ie. 6,371,008.8 meters.

So it does not mater what ellipsoid model was chosen in the end, because whatever the model chosen, it always gets normalized to a unit sphere before being scaled to that arbitrary dimension.

Reply:
![image](https://github.com/user-attachments/assets/4c296d3d-2649-4934-a8b2-f22d2a41d635)
Here's a small sketch I drew to illustrate my point.

Reply:
I still believe there's something wrong related to taking into account ellipsoid characteristics. The value of scale factors compared to other projections seems to be totally off:
```
$ echo 2 49 | bin/proj +proj=merc -S 
222638.98	6242596.00	<1.52134 1.52134 2.31449 0 1.52134 1.52134>

$ echo 2 49 | bin/proj +proj=tmerc -S 
146339.48	5431555.61	<1.00026 1.00026 1.00053 0 1.00026 1.00026>

 echo 2 49 | bin/proj +proj=s2 -S 
0.53	1.05	<0.823654 1.0871 0.895218 15.8914 1.08739 0.823271>

$ echo 2 49 | bin/proj +proj=airocean -S 
13269980.92	20976603.56	<5.70685e+06 6.08587e+06 3.46329e+13 5.67366 6.18381e+06 5.60058e+06>
```